ipn/backend.go

```diff
@@ -156,4 +157,7 @@ type Backend interface {
 	// make sure they react properly with keys that are going to
 	// expire.
 	FakeExpireAfter(x time.Duration)
+	// Ping attempts to start connecting to the given IP and sends a Notify
+	// with its PingResult.
+	Ping(ip string)
```

Comment:
Document that this has an internal timeout, so callers can expect to always receive a PingResult for every Ping.

Reply:
Added a comment that it does not and that the timeout is in the CLI, and then added a timeout to the CLI.